apps/monograph/app/components/monographpost/index.tsx

```diff
@@ -90,6 +90,13 @@ function generateTableOfContents() {
         : level;
     currentHeading = headingLevel;
 
+    const isCalloutHeading =
+      !heading.previousSibling &&
+      heading.parentElement?.classList.contains("callout");
+    if (isCalloutHeading) {
```

Comment:
You should do this at the very start of the loop.

Reply:
If we move this to the start of the loop, heading levels aren't respected for headings inside callout. 

Check at start of loop: 
![image](https://github.com/user-attachments/assets/a08a9dae-e8f1-444b-9bbb-a15e03084fc7)


Otherwise:
![image](https://github.com/user-attachments/assets/c183ea63-de1e-47eb-a72d-e6ff6af5722e)

Reply:
^ ignore